### PR TITLE
Add direction indicator to the location marker

### PR DESCRIPTION
**Pull request by nirvn**
This PR adds a direction indicator to the GPS location marker (another UX that's expected by most users):
Peek 2019-12-26 14-13
_Look for the discrete animation indicating the "live" nature of the GPS location marker_

As can be seen in the screenshot, the PR also reworks the looks of the location marker (as well as the position information panel). The motivation here is to associate the light blue on the GPS button to the on-canvas GPS location, helping users better understand what they are looking at.

@m-kuhn , the PR also fixes the locale issue in the position information overlay panel we discussed about a few days ago.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-99de3fae7e127184475dab1242aceb7fe0a7c286-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-99de3fae7e127184475dab1242aceb7fe0a7c286-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-99de3fae7e127184475dab1242aceb7fe0a7c286-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-99de3fae7e127184475dab1242aceb7fe0a7c286-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-a76eec2090b217702361b2443e6ec6402a7dc3f6-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-a76eec2090b217702361b2443e6ec6402a7dc3f6-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-a76eec2090b217702361b2443e6ec6402a7dc3f6-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-a76eec2090b217702361b2443e6ec6402a7dc3f6-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-ebef4e32aa7c8e313b65587e6cfac615d394cd51-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-ebef4e32aa7c8e313b65587e6cfac615d394cd51-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-ebef4e32aa7c8e313b65587e6cfac615d394cd51-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-ebef4e32aa7c8e313b65587e6cfac615d394cd51-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-4eac64b70b7c29c1679595f928d1b1a3c83c255a-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-4eac64b70b7c29c1679595f928d1b1a3c83c255a-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-4eac64b70b7c29c1679595f928d1b1a3c83c255a-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-4eac64b70b7c29c1679595f928d1b1a3c83c255a-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-a1325ecce7f07bdc1a01e55cde7c4fdea88b66cd-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-a1325ecce7f07bdc1a01e55cde7c4fdea88b66cd-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-a1325ecce7f07bdc1a01e55cde7c4fdea88b66cd-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-a1325ecce7f07bdc1a01e55cde7c4fdea88b66cd-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-c9a8f95218cafe7534d2efd01aaf3a13f02cfd39-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-c9a8f95218cafe7534d2efd01aaf3a13f02cfd39-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-c9a8f95218cafe7534d2efd01aaf3a13f02cfd39-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-c9a8f95218cafe7534d2efd01aaf3a13f02cfd39-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-0c63ef10124a14158ab968b4c4097a4d033d6668-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-0c63ef10124a14158ab968b4c4097a4d033d6668-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-0c63ef10124a14158ab968b4c4097a4d033d6668-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-0c63ef10124a14158ab968b4c4097a4d033d6668-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-efa5807c5a9b12ecb02fd2179ff06dd31e07363e-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-efa5807c5a9b12ecb02fd2179ff06dd31e07363e-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-efa5807c5a9b12ecb02fd2179ff06dd31e07363e-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-efa5807c5a9b12ecb02fd2179ff06dd31e07363e-x86_64.apk)

File: src/qml/PositionInformationView.qml [hunk 1]
```diff
@@ -72,14 +72,14 @@ Rectangle {
...
+            altitude = Number( positionSource.projectedPosition.z ).toLocaleString( Qt.locale(), 'f', 2 )
             if ( !isNaN( parseFloat( antennaHeight ) ) ) {
-              altitude += '<font color="#2f2f2f"><i>(%1)</i></font>'.arg((antennaHeight > 0 ? "+" : "") + Math.abs(antennaHeight).toFixed(2))
+              altitude += ' <font color="#2f2f2f"><i>(%1)</i></font>'.arg((antennaHeight > 0 ? "+" : "") + Math.abs(antennaHeight).toLocaleString(Qt.locale(), 'f', 2))
             }
           }
           else
             altitude = qsTr( "N/A" )
-          altitude = qsTr( "Altitude" ) + ': ' + altitude
+          altitude = qsTr( "Altitude" ) + ': ' + altitude + 'm'
```
Comment (m-kuhn) [inline src/qml/PositionInformationView.qml]:
```suggestion
          altitude = qsTr( "Altitude: %1 m" ).arg( altitude )
```

Comment (nirvn) [reply src/qml/PositionInformationView.qml]: @m-kuhn , thanks fixed.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-2bdc2b276fcac99ae2fb5120881a142ae4d0f68b-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-2bdc2b276fcac99ae2fb5120881a142ae4d0f68b-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-2bdc2b276fcac99ae2fb5120881a142ae4d0f68b-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-757-2bdc2b276fcac99ae2fb5120881a142ae4d0f68b-x86_64.apk)

**Comment by m-kuhn** [comment]
Sweet.

Did you try to use a triangle with constant width lines too? The current one [looks a bit like a compass. Or was it a satellite ;)](https://github.com/OSGeo/gdal/issues/2117#issuecomment-567314256)?

**Comment by nirvn** [comment]
I did, this looked cooler. And funnily enough, it's only after the fact that I got to realize link to the gdal proposal :)

**Comment by nirvn** [comment]
Re white outline, I meant it that way to offer some visual link to the gps button, and make sure the dot is never lost against the underlying map. I like that the overall symbol is relatively small, I feel it conveys "precision" :)

**Comment by m-kuhn** [comment]
Do we need to update the SDK for true north invokable?

**Comment by nirvn** [comment]
@m-kuhn , yep. If you are OK with it, I'd like to get this merged, then add the true north stuff in a different PR.

**Comment by rjwillson** [comment]
is there any way to make the direction marker permanent and to force it to use the devices internal compass rather than using the satellites to determine what direction one is heading when moving. I have seen apps that will switch automatically to movement (rather than internal compass) when the device's location moves faster than a set speed so that the compass can change to using satellites when moving in a vehicle because when in a vehicle the internal compass doesn't work properly

under the current implementation the directional arrow disappears when you stop moving. This makes it difficult to see the direction you are facing when you stop. And of course the direction is based on the satellites so based on movement. This makes it difficult to use navigate or move around in the bush/field without wiping out trying to look at phone while walking.

**Comment by m-kuhn** [comment]
Yes, that would be possible.

**Pull request merged**